### PR TITLE
[DataProcessor] Move image_processor to unified directory and add MultiModalProcessor

### PR DESCRIPTION
**Pull request by luukunn**
## Motivation

本 PR 旨在对多模态（VL）相关的 image processor 做统一目录管理与架构优化：

1. **统一目录结构**：将原先分散在各 VL 子目录（`qwen_vl_processor/`、`qwen3_vl_processor/`、`paddleocr_vl_processor/`、`ernie4_5_vl_processor/`）下的 `image_processor` / `image_preprocessor` 逐步迁移到统一的 `fastdeploy/input/image_processors/` 目录下，降低代码碎片化程度，便于后续维护和扩展新模型的 image processor。
2. **新增 MultiModalProcessor**：引入 `MultiModalProcessor` 作为多模态场景下的统一入口，封装 VL 请求处理流程与模型类型分发逻辑，由 `InputPreprocessor` 统一调用，替代原先在 `preprocess.py` 中按模型类型分别创建各 VL processor 的冗余代码。
3. **保留向后兼容**：在各旧路径下保留兼容导入入口（shim），确保已有的外部引用不会因迁移而中断。

### 新增文件（6 个）

| 文件路径 | 说明 |
|---|---|
| `fastdeploy/input/multimodal_processor.py` | 新增 `MultiModalProcessor` 统一多模态 processor，封装 VL 请求处理流程与模型类型分发 |
| `fastdeploy/input/image_processors/adaptive_processor.py` | 从 `ernie4_5_vl_processor/image_preprocessor/` 迁移 `AdaptiveImageProcessor` 实现 |
| `fastdeploy/input/image_processors/qwen_processor.py` | 从 `qwen_vl_processor/` 迁移 QwenVL `ImageProcessor` 实现 |
| `fastdeploy/input/image_processors/qwen3_processor.py` | 从 `qwen3_vl_processor/` 迁移 Qwen3VL `ImageProcessor` 实现 |
| `fastdeploy/input/image_processors/paddleocr_processor.py` | 从 `paddleocr_vl_processor/` 迁移 PaddleOCRVL `ImageProcessor` 实现 |
| `tests/input/test_multimodal_processor.py` | 新增 `MultiModalProcessor` 单元测试 |

### 兼容层（旧路径转发，4 处修改）

| 旧路径 | 转发到新路径 |
|---|---|
| `fastdeploy/input/qwen_vl_processor/image_processor.py` | → `image_processors.qwen_processor` |
| `fastdeploy/input/qwen3_vl_processor/image_processor.py` | → `image_processors.qwen3_processor` |
| `fastdeploy/input/paddleocr_vl_processor/image_processor.py` | → `image_processors.paddleocr_processor` |
| `fastdeploy/input/ernie4_5_vl_processor/image_preprocessor/*` | → `image_processors.adaptive_processor` |

### 其他修改（3 处）

- **`fastdeploy/input/image_processors/__init__.py`** — 更新导出，汇总新增的 image processor 模块
- **`fastdeploy/input/preprocess.py`** — 多模态分支统一改为创建 `MultiModalProcessor`，移除原先按模型类型分别创建各 VL processor 的冗余逻辑
- **`tests/input/test_image_preprocessor_adaptive.py`** — 适配 patch 路径到新模块位置
## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/input/multimodal_processor.py [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+            logits_processors_args = self._update_thinking_prompt_state(
+                request["prompt_token_ids"], request.get("logits_processors_args") or {}
+            )
+            request["logits_processors_args"] = logits_processors_args
+
+        max_tokens = max_model_len - len(request["prompt_token_ids"])
+        if request.get("max_tokens") is None:
+            request["max_tokens"] = max(1, max_tokens)
+        else:
+            request["max_tokens"] = min(max_tokens, request["max_tokens"])
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/multimodal_processor.py:300]: 调用时max_model_len不会为空

File: fastdeploy/input/preprocess.py [hunk 2]
```diff
@@ -106,74 +106,34 @@ def create_processor(self):
...
+                self.processor = MultiModalProcessor(
+                    model_name_or_path=self.model_name_or_path,
+                    model_type=model_type,
+                    config=self.model_config,
+                    limit_mm_per_prompt=self.limit_mm_per_prompt,
+                    mm_processor_kwargs=self.mm_processor_kwargs,
+                    reasoning_parser_obj=reasoning_parser_obj,
+                    tool_parser_obj=tool_parser_obj,
+                    enable_processor_cache=self.enable_processor_cache,
+                )
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/preprocess.py:122]: 已补充单测

File: fastdeploy/input/image_processors/adaptive_processor.py [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+
+    def set_pixels(self, min_pixels=None, max_pixels=None, msg=""):
+        """设定pixels"""
+        if min_pixels is not None:
+            assert isinstance(min_pixels, int) and min_pixels >= 0, "min_pixels must be positive int"
+            data_processor_logger.info(f"{msg} AdaptiveImageProcessor set min_pixels = {min_pixels}")
+            self.min_pixels = min_pixels
+            self.size["min_pixels"] = int(min_pixels)
+        if max_pixels is not None:
+            assert isinstance(max_pixels, int) and max_pixels > 0, "max_pixels must be positive int"
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/adaptive_processor.py:202]: 不会使用Python -O

File: fastdeploy/input/preprocess.py [hunk 4]
```diff
@@ -91,54 +91,34 @@ def create_processor(self):
...
-                        config=self.model_config,
-                        model_name_or_path=self.model_name_or_path,
-                        limit_mm_per_prompt=self.limit_mm_per_prompt,
-                        mm_processor_kwargs=self.mm_processor_kwargs,
-                        reasoning_parser_obj=reasoning_parser_obj,
-                        enable_processor_cache=self.enable_processor_cache,
-                    )
+                    model_type = QWEN3_VL
                 else:
                     raise ValueError(f"Unsupported model processor architecture: {architecture}. ")
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/preprocess.py:111]: 已补充单测

File: fastdeploy/input/image_processors/qwen_processor.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+                image = Image.fromarray(image)
+                image = resize(
+                    image,
+                    size=(resized_height, resized_width),
+                    resample=resample,
+                    data_format=input_data_format,
+                )
+
+            if do_rescale and do_normalize:
+                # Adjust mean and std for combined rescale+normalize
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/qwen_processor.py:195]: 不需要修复

File: fastdeploy/input/multimodal_processor.py [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+            vocab_file_names = ["tokenizer.model", "spm.model", "ernie_token_100k.model"]
+            for name in vocab_file_names:
+                if os.path.exists(os.path.join(self.model_name_or_path, name)):
+                    Ernie4_5Tokenizer.resource_files_names["vocab_file"] = name
+                    break
+            tokenizer = Ernie4_5Tokenizer.from_pretrained(self.model_name_or_path)
+        else:
+            from paddleformers.transformers import AutoTokenizer
+
+            tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, padding_side="left", use_fast=True)
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/multimodal_processor.py:122]: FD_USE_HF_TOKENIZER环境变量只在text分支生效

File: fastdeploy/input/image_processors/paddleocr_processor.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+def make_batched_images(images) -> List[List[ImageInput]]:
+    """
+    Accepts images in list or nested list format, and makes a list of images for preprocessing.
+
+    Args:
+        images (`Union[List[List[ImageInput]], List[ImageInput], ImageInput]`):
+            The input image.
+
+    Returns:
+        list: A list of images.
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/paddleocr_processor.py]: 修复

File: tests/input/test_multimodal_processor.py [hunk 8]
```diff
@@ -0,0 +1,1132 @@
...
+
+    def test_unsupported_model_type_raises(self):
+        """Line 86: unsupported model_type should raise ValueError."""
+        with self.assertRaises(Exception):
+            # We need to let __init__ run the model_type check.
+            # Mock the parts that come after the check to isolate it.
+            with patch.object(MultiModalProcessor, "__init__", wraps=MultiModalProcessor.__init__) as _:
+                proc = object.__new__(MultiModalProcessor)
+                # Call the real __init__ which should fail on model_type check
+                MultiModalProcessor.__init__(proc, "/mock", model_type="unsupported_type")
```
[the remark this answers is not in the document]
Comment (luukunn) [reply tests/input/test_multimodal_processor.py]: 修复

File: fastdeploy/input/image_processors/adaptive_processor.py [hunk 9]
```diff
@@ -0,0 +1,527 @@
...
+                )
+                pixel_values.extend(patches)
+                vision_grid_thws.append(video_grid_thw)
+            pixel_values = np.array(pixel_values)
+            vision_grid_thws = np.array(vision_grid_thws)
+
+            data = {
+                "pixel_values_videos": pixel_values,
+                "video_grid_thw": vision_grid_thws,
+            }
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/adaptive_processor.py]: 已修复

File: fastdeploy/input/image_processors/adaptive_processor.py [hunk 10]
```diff
@@ -0,0 +1,527 @@
...
+
+        return BatchFeature(data=data, tensor_type=return_tensors)
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/adaptive_processor.py:511]: 已修复

File: fastdeploy/input/image_processors/adaptive_processor.py [hunk 11]
```diff
@@ -0,0 +1,524 @@
...
+"""
+
+"""image preprocessor adaptive"""
+
+from typing import List, Optional, Union
+
+import numpy as np
+import paddle
+import PIL
+from paddleformers.transformers.feature_extraction_utils import BatchFeature
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/adaptive_processor.py:24]: 暂不修复

File: fastdeploy/input/image_processors/qwen3_processor.py [hunk 12]
```diff
@@ -0,0 +1,333 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+
+from typing import List, Optional, Union
+
+import numpy as np
+import paddle
+import PIL
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/qwen3_processor.py:21]: 暂不修复

File: fastdeploy/input/image_processors/qwen_processor.py [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+
+from typing import List, Optional, Union
+
+import numpy as np
+import paddle
+import PIL
```
[the remark this answers is not in the document]
Comment (luukunn) [reply fastdeploy/input/image_processors/qwen_processor.py:21]: 暂不修复

**Review by LiqinruiG — APPROVED** [review]
LGTM
